onnx/optimizer/passes/fuse_conv_add_into_bias.h

```diff
@@ -0,0 +1,43 @@
+// ATTENTION: The code in this file is highly EXPERIMENTAL.
+// Adventurous users should note that the APIs will probably change.
+
+#pragma once
+
+#include "onnx/optimizer/passes/optimize_pass.h"
+
+namespace ONNX_NAMESPACE { namespace optimization {
+
+struct FuseConvAddIntoBias final : public OptimizePass {
+  explicit FuseConvAddIntoBias()
+    : OptimizePass("fuse_conv_add_into_bias", API_TYPE::IR) {
+  }
+
+  static int idx_of_conv(const onnx::ArrayRef<onnx::Value *> & values) {
+    for (size_t i = 0; i < values.size(); i++)
+      if (values[i]->node()->kind() == kConv)
+        return i;
+    return -1;
+  }
+
+  void fuse_conv_add_into_bias(Graph& graph) {
+    for (auto it = graph.begin(); it != graph.end(); ++it) {
+      auto* n = *it;
+      if (n->kind() == kAdd) {
+        int idx = idx_of_conv(n->inputs());
+        if (idx != -1 && n->inputs()[idx]->node()->inputs().size() <= 2) {
```

Comment:
We need to check 1) Add's broadcast flag, 2) the shape of bias. (The first dimension of bias is the same as the Conv's 2nd input's 1st dimension.)

Reply:
I will check whether there is a bug in our IR constructor or just we don't have enough information in this case.

Reply:
Ok. I’ll also check the constructor.

Reply:
@houseroad 
For priori,
```
NodeProto doesn't have dim information. And field type of input and output is string.
TensorProto and ValueInfoProto has dim information.
```

In my practice,
I call `def make_graph(nodes, name, inputs, outputs, initializer=None, doc_string=None)` to make graph.

nodes: list of NodeProto
inputs: list of ValueInfoProto
outputs: list of ValueInfoProto
initializer: list of TensorProto

I think it should be a problem that each node does not keep the output shape.
onnx knows shape information of inputs and outputs and initializer.
But after one operator, shape information disappears. We can not find it, because it doesn't exist in graph.

Reply:
yes, there is no guarantee to have size information for intermediate values. We are separately working on adding shape inference, to fill it in in more cases.